django/utils/functional.py

```diff
@@ -128,6 +128,19 @@ def __cast(self):
             else:
                 return func(*self.__args, **self.__kw)
 
+        if six.PY3:
+            def __str__(self):
+                return six.text_type(self.__cast())
```

Comment:
I did that, but flake8 was complaining about `unicode` on Python3, so I went for consistency across all methods. I deliberately didn't use `python_2_unicode_compatible`, since it will always call `__unicode__()` on the proxied object, even when casting to `str()`. If there are subtle differences between `__unicode__()` and `__str__()` on the original class, I don't want those to be lost.